src/test/java/mate/academy/dao/impl/UserDaoTest.java

```diff
@@ -0,0 +1,87 @@
+package mate.academy.dao.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoTest extends AbstractTest {
+    private UserDao userDao;
+    private RoleDao roleDao;
+    private User bob;
+    private Role role;
+
+    @BeforeEach
+    void setUp() {
+        role = new Role(Role.RoleName.ADMIN);
+        userDao = new UserDaoImpl(getSessionFactory());
+        roleDao = new RoleDaoImpl(getSessionFactory());
```

Comment:
do we need to do this Before Each test method? Lets initialize services in BeforeAll (fix in all such places)